### PR TITLE
Fixes #42 

### LINKED ISSUE #42: Create an addon to run Che on Minishift
Refer: https://github.com/minishift/minishift/issues/1264

cc @l0rd  @budhrg @gbraad

### PR DESCRIPTION
**Pull request by kameshsampath**
Fixes #42 

Enhancements from #43 by @sunix

* Created image streams for Che in namespace `openshift`
* Modified template to use image streams 
* Creates Che as OpenShift template that can be used across projects

### CODE REVIEW

File: add-ons/che/LICENSE [hunk 1]
```diff
@@ -0,0 +1,201 @@
+                                 Apache License
```
Comment (budhrg) [inline add-ons/che/LICENSE]: Is it required?

File: add-ons/che/README.adoc [hunk 2]
```diff
@@ -0,0 +1,26 @@
+= Eclipse Che Addon
+
+This addon create Eclipse Che templates and image streams on OpenShift. In short this helps in setting up Eclipse Che
```
Comment (budhrg) [inline add-ons/che/README.adoc]: Probably start with

```
This addon helps in setting up Eclipse Che on Minishift inline with ..... and creats Eclipse Che templates and image streams on OpenShift. 
```

**Comment by budhrg** [comment]
@kameshsampath , does it mean #43 needs to be closed?

File: add-ons/che/LICENSE (hunk 1 above)
Comment (gbraad) [reply add-ons/che/LICENSE]: This could be a general statement... But it is right to discuss, as now the addons are public domain?

Comment (kameshsampath) [reply add-ons/che/LICENSE]: I usually add it as practice, if you guys done require it will remove it

File: add-ons/che/README.adoc (hunk 2 above)
Comment (kameshsampath) [reply add-ons/che/README.adoc]: thanks, the README is just a placeholder am @sunix is till reviewing it and need to add one more stuff to template (WIP) once it done will polish the README

File: add-ons/che/LICENSE (hunk 1 above)
Comment (budhrg) [reply add-ons/che/LICENSE]: @kameshsampath Yes, we have it at project level https://github.com/minishift/minishift-addons/blob/master/LICENSE

Comment (gbraad) [reply add-ons/che/LICENSE]: Unnecessary to state again

Comment (LalatenduMohanty) [reply add-ons/che/LICENSE]: @kameshsampath We have the license for the project so you do not need another copy of the same license.

File: add-ons/che/README.adoc (hunk 2 above)
Comment (LalatenduMohanty) [reply add-ons/che/README.adoc]: I will suggest just ` This addon sets up Eclipse Che on Minishift` is enough. Not sure why we are saying it create Eclipse Che templates and image streams as we are adding the required templates and image-streams but not creating them.

**Comment by kameshsampath** [comment]
@budhrg - I guess we can close #43  and @sunix can share his feedback over here on this issue

**Comment by sunix** [comment]
hello, So this addon is installing a che-server template. However, user would have to create and instantiate che afterward. I'm not sure this is the "spirit" of a minishift addon. I expect from an addon that it deploys che-server and that to be ready to be used right away. Am I wrong ?

**Comment by LalatenduMohanty** [comment]
> hello, So this addon is installing a che-server template. However, user would have to create and instantiate che afterward. I'm not sure this is the "spirit" of a minishift addon. I expect from an addon that it deploys che-server and that to be ready to be used right away.

You are right, the expectation is after the add-on is applied user should be able to use Che.

**Comment by LalatenduMohanty** [comment]
As discussed with @kameshsampath on #minishft IRC I am closing this as you have originally based it on #42 and @sunix has re-opened #42

**Pull request closed without merging**